session.go

```diff
@@ -459,7 +485,15 @@ func (s *Session) mux(remoteBegin *frames.PerformBegin) {
 				}
 				link, ok := links[body.Handle]
 				if !ok {
-					// TODO: per section 2.8.17 I think this should return an error
+					select {
+					case s.close <- &Error{
+						Condition:   ErrCondUnattachedHandle,
+						Description: "received transfer frame referencing a handle that's not in use",
+					}:
+						s.doneErr = fmt.Errorf("received transfer frame with unknown link handle %d", body.Handle)
+					default:
+						// close error already pending
```

Comment:
Let's log here - if we ever see this show up in our normal use that'd be huge.

Reply:
(there's another one below, same deal)

Reply:
Done.  I also consolidated all of these `selects` into a helper func.